From d48fe33782baea82af9e445fe3fc21eaa22e2f34 Mon Sep 17 00:00:00 2001
From: Jonathan Wakely <jwakely@redhat.com>
Date: Tue, 9 Feb 2021 11:23:29 +0000
Subject: [PATCH 4/4] libstdc++: Make
 coroutine_handle<_Promise>::from_address() noexcept [PR 99021]

The coroutine_handle<void>::from_address(void*) version is already
noexcept, and they do the same thing. Make them consistent.

libstdc++-v3/ChangeLog:

	PR libstdc++/99021
	* include/std/coroutine (coroutine_handle<P>::from_address): Add
	noexcept.
---
 libstdc++-v3/include/std/coroutine | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libstdc++-v3/include/std/coroutine b/libstdc++-v3/include/std/coroutine
index 2b635b982e0..f0a8bdb6840 100644
--- a/libstdc++-v3/include/std/coroutine
+++ b/libstdc++-v3/include/std/coroutine
@@ -206,7 +206,7 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
 
       constexpr void* address() const noexcept { return _M_fr_ptr; }
 
-      constexpr static coroutine_handle from_address(void* __a)
+      constexpr static coroutine_handle from_address(void* __a) noexcept
       {
 	coroutine_handle __self;
 	__self._M_fr_ptr = __a;
-- 
2.31.0.windows.1

